drydock/manifest_builder/infrastructure/tutor_based_manifests.py

```diff
@@ -0,0 +1,93 @@
+    """
+
+    def __init__(self, options: dict) -> None:
+        """Initialize the class based on the `manifest_options` from the reference.
+
+        Parameters
+        ----------
+        options: dict
+            Defines additional configuration options for the generations of the
+            configuration repository.
+
+            - ["output"]: Name of the directory to store the manifests.
+        """
+        self.options = options
+
+    def render(self, root: str, config: DrydockConfig) -> None:
+        """Register drydock custom templates and render a tutor env."""
+        with hooks.Contexts.APP("drydock-base").enter():
+            hooks.Filters.ENV_TEMPLATE_ROOTS.add_item(pkg_resources.resource_filename("drydock", "templates"))
+            hooks.Filters.ENV_TEMPLATE_TARGETS.add_items(
+                [
+                    ("default/base", "drydock"),
+                    ("default/extensions", "drydock"),
+                    ("default/kustomization.yml", "drydock"),
+                ],
+            )
+        tutor_env.save(root, config.get_data())
+        hooks.Filters.ENV_TEMPLATE_ROOTS.clear(context=hooks.Contexts.APP("drydock-base").name)
+        hooks.Filters.ENV_TEMPLATE_TARGETS.clear(context=hooks.Contexts.APP("drydock-base").name)
+
+    def relocate_env(self, src: str, dst: str) -> None:
+        """Moves the drydock rendered templates and tutor plugins to src.
+
+        At the moment we render a full tutor env with our templates under
+        'env/drydock/default'. The final drydock env is the one in 'env/drydock'
+        **plus** the plugin directory that has to be manually relocated to
+        'env/drydock/base/plugins'.
+
+        Parameters
+        ----------
+        src: str
+            The inital path where the full tutor env was rendered.
+        dst: str
+            The path to save the final drydock env.
+        """
+        base_dir = path_join(src, "env/drydock/default")
+        plugins_dir = path_join(src, "env/plugins")
+        shutil.move(base_dir, dst)
+        if os.path.exists(plugins_dir):
+            shutil.move(plugins_dir, path_join(dst, "base"))
+
+    def save(self, config: DrydockConfig) -> None:
+        """Creates an alternative tutor env with the base openedx installation.
+
+        The generated env consists of a Kustomize application that uses the original
+        tutor env as a base in addition to an `extensions` overlay to include additional
+        resources.
+
+        Parameters
+        ----------
+        config: DrydockConfig
+            A Tutor configuration extension.
+        """
+        tutor_root = config.get_root()
+        dst = path_join(tutor_root, self.options.get("output", "drydock-env"))
+        with tempfile.TemporaryDirectory() as tmpdir:
+            src = path_join(tmpdir)
+            self.render(src, config)
+            shutil.rmtree(path=dst, ignore_errors=True)
+            self.relocate_env(src, dst)
```

Comment:
This is my current folder structure:
```
├── config.yml
├── env
│   ├── base
│   │   ├── apps
│   │   │   ├── caddy
│   │   │   ├── openedx
│   │   │   └── redis
│   │   ├── k8s
│   │   │   ├── deployments.yml
│   │   │   ├── jobs.yml
│   │   │   ├── namespace.yml
│   │   │   ├── services.yml
│   │   │   └── volumes.yml
│   │   ├── kustomization.yml
│   │   └── plugins
│   │       ├── forum
│   │       ├── notes
│   │       └── settingmanager.yml
│   ├── extensions
│   │   ├── flowers.yml
│   │   ├── hpa.yml
│   │   ├── ingress.yml
│   │   ├── kustomization.yml
│   │   └── overrides.yml
│   └── kustomization.yml
├── reference.yml
└── requirements.txt
```
My tutor plugins list:
```
android         	(disabled)	13.0.0
discovery       	(disabled)	13.0.1
distro          	          	2.1.0
drydock         	          	0.1.0
ecommerce       	(disabled)	13.0.1
forum           	          	13.0.0
license         	(disabled)	13.0.0
mfe             	(disabled)	13.0.6
minio           	          	13.0.1
notes           	          	13.0.1
richie          	(disabled)	13.0.2
settingmanager  	          	0.1.0
webui           	(disabled)	13.0.2
xqueue          	(disabled)	13.0.0
```
Then I run: 
`tutor drydock save -r reference.yml`
When the command finishes, I run:
`tutor plugins list`
And this happens:
```
⚠️  Failed to enable plugin 'settingmanager': plugin 'settingmanager' is not installed.
android    	(disabled)	13.0.0
discovery  	(disabled)	13.0.1
distro     	          	2.1.0
drydock    	          	0.1.0
ecommerce  	(disabled)	13.0.1
forum      	          	13.0.0
license    	(disabled)	13.0.0
mfe        	(disabled)	13.0.6
minio      	          	13.0.1
notes      	          	13.0.1
richie     	(disabled)	13.0.2
webui      	(disabled)	13.0.2
xqueue     	(disabled)	13.0.0
```
Is that okay?

Reply:
Yes (actually is not ok, but is a current limitation). At the moment we don't render module nor yaml plugins on `base/plugins`. The current workaround is to define the `TUTOR_PLUGINS_ROOTS` outside `env`.

The folder structure would look something like this:

```
├── config.yml
├── env
│   ├── base
│   │   ├── apps
│   │   │   ├── caddy
│   │   │   ├── openedx
│   │   │   └── redis
│   │   ├── k8s
│   │   │   ├── deployments.yml
│   │   │   ├── jobs.yml
│   │   │   ├── namespace.yml
│   │   │   ├── services.yml
│   │   │   └── volumes.yml
│   │   ├── kustomization.yml
│   │   └── plugins
│   │       ├── forum
│   │       └──  notes
│   ├── extensions
│   │   ├── flowers.yml
│   │   ├── hpa.yml
│   │   ├── ingress.yml
│   │   ├── kustomization.yml
│   │   └── overrides.yml
│   └── kustomization.yml
├── reference.yml
└── requirements.txt
└── plugins
    └── settingmanager.yml
```

and your `TUTOR_PLUGINS_ROOT` should be something like `$TUTOR_ROOT/plugins`